test/src/main/java/com/redhat/lightblue/test/AbstractCRUDTestController.java

```diff
@@ -0,0 +1,122 @@
+/*
+ Copyright 2015 Red Hat, Inc. and/or its affiliates.
+
+ This file is part of lightblue.
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.redhat.lightblue.test;
+
+import static com.redhat.lightblue.util.JsonUtils.json;
+import static com.redhat.lightblue.util.test.AbstractJsonNodeTest.loadJsonNode;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+
+import org.junit.AfterClass;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.redhat.lightblue.Request;
+import com.redhat.lightblue.config.DataSourcesConfiguration;
+import com.redhat.lightblue.config.JsonTranslator;
+import com.redhat.lightblue.config.LightblueFactory;
+import com.redhat.lightblue.metadata.EntityMetadata;
+import com.redhat.lightblue.metadata.Metadata;
+
+/**
+ * Testing harness for junit tests that need to stand an in-memory instance of lightblue.<br>
+ * <b>NOTE:</b> This does not use the rest layer.
+ *
+ * <p>
+ * Example Usage:<br>
+ * <code>
+ * Response response = lightblueFactory.getMediator().insert(
+ * createRequest_FromResource(InsertionRequest.class, "./path/to/insert/metadata.json"));
+ * </code></p>
+ *
+ * @author dcrissman
+ */
+public abstract class AbstractCRUDTestController {
+
+    protected static LightblueFactory lightblueFactory;
+
+    @AfterClass
+    public static void cleanup(){
+        lightblueFactory = null;
+    }
+
+    /**
+     * Creates an instance of the {@link LightblueFactory}.<br>
+     * <b>Must be called from implementations in a static BeforeClass method</b>
+     * @param datasourcesResourcePath - path to datasources.json
+     * @param metadataResourcePaths - path to any and all *-metadata.json you'd like {@link LightblueFactory} to know about
+     * @return an instance of {@link LightblueFactory}.
+     * @throws IOException
+     * @throws ClassNotFoundException
+     * @throws NoSuchMethodException
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     * @throws InstantiationException
+     */
+    protected static LightblueFactory createLightblueFactory(String datasourcesResourcePath, String... metadataResourcePaths)
```

Comment:
Should this be executed by some @BeforeClass annotated method?  Could make the arguments abstract getter methods.  This way the factory is always created.